contrib/buildrpm/buildrpmLibfabric.sh

```diff
@@ -0,0 +1,362 @@
+         ;;
+      e) configure_options="$configure_options --disable-$OPTARG"
+         ;;
+      d) configure_options="$configure_options --enable-debug"
+         ;;
+      c) configure_options="$configure_options $OPTARG"
+         ;;
+      r) configure_options="$rpmbuild_options $OPTARG"
+         ;;
+      s) unpack_spec="true"
+         ;;
+      v) verbose="true"
+         ;;
+      h) echo "$usage"
+         exit 0
+         ;;
+    esac
+done
+shift $(( OPTIND - 1 ));
+
+########################################
+# Check if there is at least 1 parameter
+# specified (tarball is mandatory)
+########################################
+if [[ $# -lt 1 ]]; then
+  echo "$usage" 1>&2
+  exit 1
+fi
+
+#############
+# Print hello
+#############
+if [[ -n "$verbose" ]]; then
+cat <<EOF
+
+------------------------------------------------------------------------------
+====              Welcome to Libfabric RPM build script                   ====
+------------------------------------------------------------------------------
+You have specified this options:
+ $arguments
+------------------------------------------------------------------------------
+
+EOF
+  if [[ -n "$noop" ]]; then
+    echo "$0: --> Option -n was specified. I won't do anything."
+  fi
+fi
+
+################################
+# Check if tarball was specified
+################################
+export tarball="$1"
+
+if [[ ! -r "$tarball" ]]; then
+  echo "$0: FATAL: Can't find $tarball" 1>&2
+  exit 2
+fi
+verbose "Found tarball: $tarball"
+
+##############################
+# Get tarball name and version
+##############################
+export tardirname=$(basename "$tarball" | awk '{print substr ($0, 0, index($0, ".tar")-1)}')
+if [[ -z "$tardirname" ]]; then
+  echo "$0: FATAL: Cannot determine name from $tarball" 1>&2
+  exit 3
+fi
+verbose "Name of package: $tarball"
+
+export version=$(basename "$tardirname" | cut -d- -f2)
+if [[ -z "$tardirname" ]]; then
+  echo "$0: FATAL: Cannot determine version from $tarball" 1>&2
+  exit 4
+fi
+verbose "Version: $version"
+
+if [[ -n "$install_in_opt" ]]; then
+  prefix="/opt/libfabric/$version"
+  rpmbuild_options="$rpmbuild_options --define '_prefix $prefix'"
+  verbose "Setting RPM install path to: $prefix"
+fi
+
+######################################
+# Try to unpack spec file from tarball
+######################################
+if [[ -f "$specfile" ]]; then
+  if [[ -n "$verbose" ]]; then
+    echo "$0: WARNING: $specfile already exists and will be overwritten" 1>&2
+  fi
+fi
+
+if [[ -n "$unpack_spec" ]]; then
+  if [[ -z "$verbose" ]]; then
+    if [[ -z "$noop" ]]; then
+      tar -xf "$tarball"
+      cp -fp "$tardirname/$specfile" .
+      rm -rf "$tardirname"
+    fi
+  else
+    verbose "Extracting tarball"
+    runcmd "tar -xvf \"$tarball\""
+    verbose "Copying specfile"
+    runcmd "cp -fpv \"$tardirname/$specfile\" ."
+    verbose "Cleanup after extraction"
+    runcmd "rm -rfv \"$tardirname\""
+  fi
```

Comment:
I wouldn't worry about the `-v` options to `tar`, `cp`, and `rm` -- which will allow you to unify these two blocks.  I.e., you can always just

```sh
 verbose "Extracting tarball"
 runcmd "tar -xvf \"$tarball\""
 verbose "Copying specfile"
 runcmd "cp -fpv \"$tardirname/$specfile\" ."
 verbose "Cleanup after extraction"
 runcmd "rm -rfv \"$tardirname\""
```

because `verbose` and `runcmd` check the verbosity setting for you -- no need to have a separate block for it.

Additionally, the 2-block method you have right now does not check the error status of the `tar`, `cp`, and `rm` in the first block.